### PR TITLE
Issue #4367: Add pitest non-checks code profiles

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4367 

Add pitest-checkstyle-common profile for classes in com.puppycrawl.tools.checkstyle.* , 
com.puppycrawl.tools.checkstyle.ant.* , com.puppycrawl.tools.checkstyle.doclets.* packages, except Main class and classes connected with tree.
execution time 10:05 min
threshold: 78

Add pitest-checkstyle-main profile for Main class.
Contains many tests, executes better in separate profile.
execution time 06:25 min
threshold: 93

Add pitest-checkstyle-tree-walker profile for classes from the base package connected with tree. Contains many tests on TreeWalker, executes better in separate profile.
execution time:09:57 min
threshold: 83

api, filters, utils and gui profiles listed below contains many classes, it would be better to cover them separately and maybe unite in the end.

Add pitest-checkstyle-api profile for com.puppycrawl.tools.checkstyle.api.* package.
execution time: 08:04 min
threshold: 46

Add pitest-checkstyle-filters profile for com.puppycrawl.tools.checkstyle.filefilters.* and com.puppycrawl.tools.checkstyle.filters.* packages. United filters packages in one profile.
execution time: 03:10 min
threshold: 95

Add pitest-checkstyle-utils for com.puppycrawl.tools.checkstyle.utils.* package;
execution time: 03:17 min
threshold: 40

add pitest-checkstyle-gui profile for com.puppycrawl.tools.checkstyle.gui.* package;
execution time: 01:48 min
threshold: 30

com.puppycrawl.tools.checkstyle.grammars was skipped as it doesn't contain any classes (only interface).

### CODE REVIEW

**Comment by Nimfadora** [comment]
@romani, @rnveach, sevntu-checkstyle-check did not pass, but I guess it is not my fault, as without my changes build did't pass anyway.

**Comment by rnveach** [comment]
> sevntu-checkstyle-check did not pass

This is an issue in `master` and can be ignored for now. See https://github.com/checkstyle/checkstyle/issues/4387

**Comment by romani** [comment]
@Nimfadora , 

1) please explain why you split classes in such groups 
2) please provide execution time for each profile
3) please update https://github.com/checkstyle/checkstyle/blob/master/shippable.yml#L8 with your profiles as your constraints should start to work immediately after PR is applied.

**Comment by Nimfadora** [comment]
@romani, points 1 and 2 should be stated in PR first comment, am I right?
I included all classes from the main folder and all test classes from the folder. AuditEventUtFormatter, BaseCheckTestSupport, BaseFileSetCheckTestSupport, BriefUtLogger, DebugFilter, DebugAuditAdapter, TestBeforeExecutionFileFilter, TestRootModuleChecker were not included as they are in test folder, but not actually test classes.

**Comment by romani** [comment]
> points 1 and 2 should be stated in PR first comment, am I right?

it is not critical, but now I see timing and I am fine with changes.
During answers please keep reference to point numbers, to make it clear to what point you answer.

4) please resolve problem on Shippable CI
https://app.shippable.com/github/checkstyle/checkstyle/runs/2981/17/console
```
PROFILE="-Ppitest-checkstyle-api,no-validations" 
...
Failed to execute goal org.pitest:pitest-maven:1.2.0:mutationCoverage (default-cli) on project checkstyle: Mutation score of 45 is below threshold of 46 -> [Help 1
```

@rnveach , as CI is resolved , I am ready to merge.

File: pom.xml [hunk 1]
```diff
@@ -1866,6 +1866,212 @@
...
+            <configuration>
+              <targetClasses>
+                <param>com.puppycrawl.tools.checkstyle.api.*</param>
+              </targetClasses>
+              <targetTests>
+                <param>com.puppycrawl.tools.checkstyle.api.*</param>
+              </targetTests>
+              <excludedClasses>
+                <!-- deprecated class -->
+                <param>com.puppycrawl.tools.checkstyle.api.Check</param>
```
Comment (rnveach) [inline pom.xml]: Would removing this exclude cause an issue or lower our mutation threshold?
Class has absolutely no methods or fields, so I can't see how mutation wouldn't be 100% for it.

In other pitests, deprecated classes did have fields/methods and caused a lower score for missing tests on them.

**Comment by Nimfadora** [comment]
@rnveach, what about `annotation`, I think it is better to add it in another issue, as it belongs to the "check" profile. But if you insist I can recommit.

**Comment by rnveach** [comment]
> I think it is better to add it in another issue

Then please create the issue.

File: pom.xml (hunk 1 above)
Comment (Nimfadora) [inline pom.xml]: Yes, you are right. I'll remove it while increasing coverage for this profile.

**Comment by romani** [comment]
yes, please create for `annotation` package coverage separate PR (no need to make issue for such a minor update), this PR will be merged as soon Shippable is passed.

File: pom.xml (hunk 1 above)
Comment (romani) [inline pom.xml]: it will lower. By mean of such excludes we will gradually raise threshold to 100%.

**Comment by romani** [comment]
merged , issue for annotations - #4402

File: pom.xml (hunk 1 above)
Comment (Nimfadora) [inline pom.xml]: @romani, but it does not lower the coverage, according to recent check. So if needed I can remove this exclude.

Comment (romani) [inline pom.xml]: Please remove , the less the config the better.

**Pull request merged**